plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java

```diff
@@ -458,8 +459,13 @@ private DeltaLakeTransactionLogEntry buildAddEntry(ConnectorSession session, Blo
         log.debug("Block %s has %s fields", block, addEntryRow.getFieldCount());
         CheckpointFieldReader add = new CheckpointFieldReader(session, addEntryRow, type);
 
-        String path = add.getString("path");
         Map<String, String> partitionValues = add.getMap(stringMap, "partitionValues");
+        Map<String, Optional<String>> canonicalPartitionValues = canonicalizePartitionValues(partitionValues);
+        if (!partitionConstraint.isAll() && !partitionMatchesPredicate(canonicalPartitionValues, partitionConstraint.getDomains().orElseThrow())) {
+            return null;
```

Comment:
Add a code comment why it's OK to return here when processing a non-null add entry.

It would be nice to enhance `io.trino.plugin.deltalake.transactionlog.checkpoint.CheckpointEntryIterator.CheckPointFieldExtractor#getEntry` javadoc to indicate when it is OK to return a null value